Mathlib/Algebra/Hom/GroupAction.lean

```diff
@@ -0,0 +1,578 @@
+protected def id : X →[M'] X :=
+  ⟨id, fun _ _ => rfl⟩
+#align mul_action_hom.id MulActionHom.id
+
+@[simp]
+theorem id_apply (x : X) : MulActionHom.id M' x = x :=
+  rfl
+#align mul_action_hom.id_apply MulActionHom.id_apply
+
+variable {M M' Z}
+
+/-- Composition of two equivariant maps. -/
+def comp (g : Y →[M'] Z) (f : X →[M'] Y) : X →[M'] Z :=
+  ⟨g ∘ f, fun m x =>
+    calc
+      g (f (m • x)) = g (m • f x) := by rw [f.map_smul]
+      _ = m • g (f x) := g.map_smul _ _⟩
+#align mul_action_hom.comp MulActionHom.comp
+
+@[simp]
+theorem comp_apply (g : Y →[M'] Z) (f : X →[M'] Y) (x : X) : g.comp f x = g (f x) :=
+  rfl
+#align mul_action_hom.comp_apply MulActionHom.comp_apply
+
+@[simp]
+theorem id_comp (f : X →[M'] Y) : (MulActionHom.id M').comp f = f :=
+  ext fun x => by rw [comp_apply, id_apply]
+#align mul_action_hom.id_comp MulActionHom.id_comp
+
+@[simp]
+theorem comp_id (f : X →[M'] Y) : f.comp (MulActionHom.id M') = f :=
+  ext fun x => by rw [comp_apply, id_apply]
+#align mul_action_hom.comp_id MulActionHom.comp_id
+
+variable {A B}
+
+/-- The inverse of a bijective equivariant map is equivariant. -/
+@[simps]
+def inverse (f : A →[M] B) (g : B → A) (h₁ : Function.LeftInverse g f)
+    (h₂ : Function.RightInverse g f) : B →[M] A
+    where
+  toFun := g
+  map_smul' m x :=
+    calc
+      g (m • x) = g (m • f (g x)) := by rw [h₂]
+      _ = g (f (m • g x)) := by rw [f.map_smul]
+      _ = m • g x := by rw [h₁]
+
+#align mul_action_hom.inverse MulActionHom.inverse
+
+end MulActionHom
+
+/-- Equivariant additive monoid homomorphisms. -/
+structure DistribMulActionHom extends A →[M] B, A →+ B
+#align distrib_mul_action_hom DistribMulActionHom
+
+/-- Reinterpret an equivariant additive monoid homomorphism as an additive monoid homomorphism. -/
+add_decl_doc DistribMulActionHom.toAddMonoidHom
+
+/-- Reinterpret an equivariant additive monoid homomorphism as an equivariant function. -/
+add_decl_doc DistribMulActionHom.toMulActionHom
+
+@[inherit_doc]
+notation:25 A " →+[" M:25 "] " B:0 => DistribMulActionHom M A B
+
+/-- `DistribMulActionHomClass F M A B` states that `F` is a type of morphisms preserving
+the additive monoid structure and scalar multiplication by `M`.
+
+You should extend this class when you extend `DistribMulActionHom`. -/
+class DistribMulActionHomClass (F : Type _) (M A B : outParam <| Type _) [Monoid M] [AddMonoid A]
+  [AddMonoid B] [DistribMulAction M A] [DistribMulAction M B] extends SMulHomClass F M A B,
+  AddMonoidHomClass F A B
+#align distrib_mul_action_hom_class DistribMulActionHomClass
+
+/- porting note: Removed a @[nolint dangerousInstance] for
+DistribMulActionHomClass.toAddMonoidHomClass not dangerous due to `outParam`s -/
+
+namespace DistribMulActionHom
+
+/- porting note: TODO decide whether the next two instances should be removed
+Coercion is already handled by all the HomClass constructions I believe -/
+-- instance coe : Coe (A →+[M] B) (A →+ B) :=
```

Comment:
```suggestion
-- instance coe : Coe (A →+[M] B) (A →+ B) :=
```

Do we still want coercion instances like these (with @[coe], if at all I guess)?

Reply:
I can foresee in case of trouble with `outParam`s we might want to re-enable these. In that case I'd prefer, if practical, an appropriate `simp` lemma rewriting it to the `HomClass` coercion, instead of a `@[coe]` annotation.